src/core/Elsa.Abstractions/Models/Variables.cs

```diff
@@ -29,10 +30,15 @@ public JToken GetVariable(string name)
             return ContainsKey(name) ? this[name] : default;
         }
 
-        public T GetVariable<T>(string name)
+        public object GetVariable(string name, string typeName)

```

Comment:
Why pass in a typeName and not a `Type` itself?

Reply:
Err.. ye... Err.... You're right 😃 I'll fix it